src/devices/Is31fl3731/Is31Fl3731.cs

```diff
@@ -0,0 +1,278 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// Represents an IS31FL3731 LED Matrix driver
+    /// </summary>
+    // Datasheet: https://cdn-learn.adafruit.com/assets/assets/000/030/994/original/31FL3731.pdf
+    // Product: https://www.adafruit.com/product/2946
+    // Product: https://www.adafruit.com/product/2974
+    public abstract class Is31fl3731 : IDisposable
+    {
+        // Register addresses
+        // Command register
+        // Points pages one to nine
+        // table 2 in datasheet
+        private const byte COMMAND_REGISTER = 0xFD;
```

Comment:
Not sure about that. It's gonna make the code look pretty ugly without target typed enums.

Let's take a look.

This line:

```charp
Write(FUNCTION_REGISTER, DISPLAY_REGISTER, (byte)value);
```

Will turn into:

```charp
Write(Registers.FUNCTION_REGISTER, Registers.DISPLAY_REGISTER, (byte)value);
```

Right? I think that makes the C# look worse as compared to the Python code I read. Seems like this is trading cleanup at the top of the file for worse readability within the core logic.

Reply:
I really prefer to use enums because they avoid problems. You can group them. And in this case have 2 different enums, one for the register and one for the command.
Short term const seems super nice. Then when it comes to maintenance, they are not good at all. Enums forces you to use the right one. Here on the Write function, I can use the wrong one very easilly.

So most likely your write will turn out like with 2 different enums. And when you have possibility to mix elements, it's much easier for maintenance and readability with enums:

```csharp
Write(Function.FUNCTION_REGISTER, Register.DISPLAY_REGISTER, (byte)value);
```